n2survey/lime/survey.py

```diff
@@ -531,6 +609,104 @@ def plot(
 
         return fig, ax
 
+    def check_plot_implemented(self, question, compare_with=None, add_questions=None):
+        """
+        function to check if question type and/or combination of questions for
+        compare_with is already implemented and working
+        """
+        if not add_questions:
+            add_questions = []
+        supported_plots = ["single-choice", "multiple-choice", "array"]
+        supported_comparisons = [("single-choice", "single-choice")]
+        all_plots = [question]
+        if compare_with:
+            all_plots.append(compare_with)
+        for i in add_questions:
+            all_plots.append(i)
+        for plot in all_plots:
+            if self.get_question_type(plot) not in supported_plots:
+                raise NotImplementedError(
+                    """
+                    Question type not yet implemented
+                    """
+                )
+        if compare_with:
+            tuple_list = [(question, compare_with)]
+            for i in add_questions:
+                tuple_list.append((question, i))
+            for i in tuple_list:
+                question_type_tuple = (
+                    self.get_question_type(i[0]),
+                    self.get_question_type(i[1]),
+                )
+                if question_type_tuple not in supported_comparisons:
+                    raise NotImplementedError(f" comparison {i} not yet implemented")
+
+    def get_answer_sequence(self, question, add_questions=None, totalbar=False):
+        """
+        creates answer sequence from given questions to keep sequence in
+        plot consistent if answers that were never chosen are suppressed
+        """
+        if not add_questions:
+            add_questions = []
+        answer_sequence = [
+            list(self.count(question, labels=True).index.values.astype(str))
```

Comment:
I want the list of answer options for each question, also the add_questions ones, each answer options has to be exactly in the same sequence as saved in the questions. 
Since I saved the combinations for question and add_questions in a list I decided to keep it the same with the answer sequences to the questions, so they are allocated at the same position in the answer_sequence list.

Yet not every answer is plotted, if an answer is given zero times in question or in the add_questions, it will not be plotted and also has to be removed from the correct answer_sequence list entry without changing the sequence.
I also decided to do all actions that require "self." to be conducted in survey.py so the plotting scripts themselves have to only deal with data and not objects handed down through survey.py to the next sublevel, the plotting script.